content/backend/graphql-js/6-authentication.md

```diff
@@ -11,52 +11,105 @@ In this section, you're going to implement signup and login functionality that a
-  links: [Link!]!
+model User {
+  id        Int      @id @default(autoincrement())
+  createdAt DateTime @default(now())
+  name      String
+  email     String   @unique
+  password  String
+  links     Link[]
 }
 ```
 
 </Instruction>
 
-You're adding a new _relation field_ called `postedBy` to the `Link` type that points to a `User` instance. The `User` type then has a `links` field that's a list of `Link`s. This is how you express a one-to-many relationship using SDL.
+Now we start to see even more how Prisma helps you to reason about your data in a way that is more aligned with how it is represented in the underlying database.
+
+### Understanding Relation Fields
+
+Notice how you're adding a new _relation field_ called `postedBy` to the `Link` model that points to a `User` instance. The `User` model then has a `links` field that's a list of `Link`s. To do this, we need to also define the relation by annotating the `postedBy` field with [the `@relation` attribute](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#the-relation-attribute). This is required for every relation field in your Prisma schema, and all you're doing is defining what the foreign key of the related table will be. So in this case, we're adding an extra field to store the `id` of the `User` who posts a `Link`, and then telling Prisma that `postedById` will be equal to the `id` field in the `User` tabel.
+
+If this is quite new to you, don't worry! We're going be adding a few of these relational fields and you'll get the hang of it as you go! For a deeper dive on relations with Prisma, check out these [docs](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations).
+
+### Updating Prisma Client
 
-After every change you're making to the datamodel file, you need to redeploy the Prisma API to apply your changes and migrate the underlying database schema.
+This is a great time to refresh your memory on the workflow we described for your project at the end of chapter 4!
+
+After every change you make to the datamodel, you need to migrate your database and then re-generate Prisma Client.
 
 <Instruction>
 
 In the root directory of the project, run the following command:
 
 ```bash(path=".../hackernews-node")
-prisma deploy
+npx prisma migrate save --experimental
+```
+
+</Instruction>
+
+<Instruction>
+
+In the root directory of the project, run the following command. When promptd, give your migration a descriptive name, such as "Add User model".
+
+```bash(path=".../hackernews-node")
+npx prisma migrate save --experimental
 ```
 
 </Instruction>
 
-This now updated the Prisma API. The Prisma client is generated automatically after running `prisma deploy`, exposing all the CRUD methods for the newly added User model.
+// TODO @nikolasburk this command actually failed here. Temporarily I had to just delete all migrations in order to move on but we should make sure that everyone isn't going to experience this issue.
```

Comment:
@nikolasburk just to point this out for you ⬆️

Reply:
Hmm, weird! What was the error message you saw?